File: Product of array elements except self/array_product_except_self.go
package main
import "fmt"

func main() {
  input := []int{1, 2, 3, 4, 5, 0}
  var totalProduct int = 1
  zeroCounter := 0
  firstZeroIndex := 0
  for i, in := range input {
      if in != 0 {
          totalProduct *= in
      }else {
          zeroCounter++
          firstZeroIndex = i
      }
  }

  output := make([]int, len(input))
  if zeroCounter == 1 {
      for i, _ := range input {
          if firstZeroIndex == i {
              output[i] = totalProduct
          }else {
              output[i] = 0
          }
      }
  } else if zeroCounter > 1 {
      for i, _ := range input {
          output[i] = 0
      }
  }else {
      for i, in := range input {
      if in == 0 {
          output[i] = 0
      }else {
          output[i] = totalProduct/in
      }
    }
  }   
  
  fmt.Printf("Result: %v", output)
}

File: Product of array elements except self/ArrayProductExceptSelf.java
import java.util.*;

class Main {
    public static void main(String[] args) {
       var input = new int[] {10, 3, 5, 6, 2};
       var output = new int[input.length];
       var zeroCounter = 0;
       var firstZeroIndex = 0;
       var totalProduct = 1;
       for(var i = 0; i < input.length; i++) {
           if (input[i] != 0) {
               totalProduct *= input[i];
           }else {
               zeroCounter++;
               firstZeroIndex = i;
           }
       }
       
       if(zeroCounter == 1) {
           for (var i = 0; i < input.length; i++) {
               if (i == firstZeroIndex) {
                   output[i] = totalProduct;
               }else {
                   output[i] = 0;
               }
           }
       } else if (zeroCounter > 1) {
           for (var i = 0; i < input.length; i++) {
               output[i] = 0;
           }
       } else {
           for (var i = 0; i < input.length; i++) {
               output[i] = totalProduct / input[i];
           }
       }
       System.out.println(Arrays.toString(output));
    }
}
